version: "3.3"
services:
  django:
    build: .
    command: python manage.py runserver 0.0.0.0:8000
    env_file:
      - .env
    extra_hosts:
      - "host.docker.internal:host-gateway"
    volumes:
      - ./:/app
    ports:
      - ${PORT_OUT}:8000
    depends_on:
      db:
        condition: service_healthy

  db:
    image: "postgres:15.0-alpine"
    environment:
      POSTGRES_DB: template_db
      POSTGRES_USER: root
      POSTGRES_PASSWORD: ludens123
    volumes:
      - ./docker/postgres:/var/lib/postgresql/data
    ports:
      - 5432:5432
    healthcheck:
      test: ["CMD-SHELL", "pg_isready --username=root --dbname=template_db"]
      interval: 5s
      timeout: 5s
      retries: 5

  # cluster:
  #   build: .
  #   command: python manage.py qcluster
  #   volumes:
  #     - ./:/app
  #   depends_on:
  #     - django

  # redis:
  #   image: "redis:5.0.3-alpine"
  #   ports:
  #     - "6379:6379"